src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java

```diff
@@ -974,6 +976,9 @@ private void debugCompleted(String tag, long startNanos, HttpRequest req) {
             }
             throw ie;
         } catch (ExecutionException e) {
+            // Exceptions are often thrown from asynchronous code, making it
+            // difficult to trace back to the user's original entry point.
+            // Wrap exceptions explicitly to preserve that context.
```

Comment:
Nit - Would it better to reword this as:

> // Exceptions are often thrown from asynchronous code, and the 
// stacktrace may not always contain the application classes.
// That makes it difficult to trace back to the application code which
// invoked the HttpClient. Here we instantiate/recreate the
// exceptions to capture the application's calling code in the
// stacktrace of the thrown exception.